testing/debugger_detection.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/testing/debugger_detection.h"
+
+#include "flutter/fml/build_config.h"
+#include "flutter/fml/logging.h"
+
+#if OS_MACOSX
+
+#include <assert.h>
+#include <stdbool.h>
+#include <sys/sysctl.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#endif  // OS_MACOSX
+
+#if OS_WIN
+
+#include <Debugapi.h>
+
+#endif  // OS_WIN
+
+#if OS_LINUX
+
+#include <sys/ptrace.h>
+
+#endif  // OS_LINUX
+
+namespace flutter {
+namespace testing {
+
+#if OS_MACOSX
```

Comment:
nit: by moving the preprocessor guards inside the definition for GetDebuggerStatus you are removing duplicate code and reducing the difference between platforms.  For example, if you changed the signature of the function while you are working on linux, you might forget to update it for mac since it won't be compiled as is.

Reply:
Done.